bin/sozo/src/ops/register.rs

```diff
@@ -1,25 +1,59 @@
+use std::collections::HashMap;
+
 use anyhow::{Context, Result};
 use dojo_world::contracts::WorldContract;
 use dojo_world::metadata::Environment;
 use dojo_world::utils::TransactionWaiter;
+use scarb::core::Config;
 use starknet::accounts::Account;
+use starknet_crypto::FieldElement;
 
 use crate::commands::register::RegisterCommand;
 
-pub async fn execute(command: RegisterCommand, env_metadata: Option<Environment>) -> Result<()> {
+pub async fn execute(
+    command: RegisterCommand,
+    env_metadata: Option<Environment>,
+    config: &Config,
+) -> Result<()> {
     match command {
         RegisterCommand::Model { models, world, starknet, account, transaction } => {
             let world_address = world.address(env_metadata.as_ref())?;
             let provider = starknet.provider(env_metadata.as_ref())?;
 
             let account = account.account(&provider, env_metadata.as_ref()).await?;
             let world = WorldContract::new(world_address, &account);
+            let mut model_class_hashes = HashMap::new();
+            for model_class_hash in models.iter() {
```

Comment:
I think the filtering below is unnecessary considering we can perform the filtering inside of this `for` loop. We can replace the `HashMap` with a simple `Vec` that stores the class hash of the models we need to register.